benchmarks/shear_bands/shear_bands.cc

```diff
@@ -748,27 +748,18 @@ namespace aspect
                       ExcMessage("Postprocessor shear bands growth rate only works with the material model shear bands."));
         }
 
-      if (dynamic_cast<const PlaneWaveMeltBandsInitialCondition<dim> *>(&this->get_compositional_initial_conditions()) != NULL)
-        {
-          const PlaneWaveMeltBandsInitialCondition<dim> *
-          initial_conditions
-            = dynamic_cast<const PlaneWaveMeltBandsInitialCondition<dim> *>(&this->get_compositional_initial_conditions());
+      const PlaneWaveMeltBandsInitialCondition<dim> *
+      initial_conditions
+      = this->get_initial_composition_manager().template find_initial_composition_model<PlaneWaveMeltBandsInitialCondition<dim> > ();
 
-          amplitude           = initial_conditions->get_wave_amplitude();
-          initial_band_angle  = initial_conditions->get_initial_band_angle();
-        }
-      else
-        {
-          AssertThrow(false,
-                      ExcMessage("Postprocessor shear bands growth rate only works with the initial conditions model "
-                                 "plane wave melt bands."));
-        }
+      amplitude           = initial_conditions->get_wave_amplitude();
```

Comment:
By getting rid of the `if(... == NULL)`, you will get a seg fault here if `initial_conditions` is NULL.